service/matching/tasklist/matcher.go

```diff
@@ -221,28 +221,34 @@ func (tm *TaskMatcher) MustOffer(ctx context.Context, task *InternalTask) error
 		return fmt.Errorf("rate limit error dispatching: %w", err)
 	}
 
+	startT := time.Now()
 	// attempt a match with local poller first. When that
 	// doesn't succeed, try both local match and remote match
 	taskC := tm.getTaskC(task)
 	select {
 	case taskC <- task: // poller picked up the task
+		tm.scope.IncCounter(metrics.AsyncMatchLocalPollCounterPerTaskList)
+		tm.scope.RecordTimer(metrics.AsyncMatchLocalPollLatencyPerTaskList, time.Since(startT))
 		return nil
 	case <-ctx.Done():
 		return fmt.Errorf("context done when trying to forward local task: %w", ctx.Err())
 	default:
 	}
 
+	attempt := 0
```

Comment:
My memory is slightly hazy here, but I'd be careful interpreting this value as the actual number of retries per task, if I recall correctly this times out (in pathological cases) and then ends up looping from `dispatchBufferedTasks(`, so the actual number of attempts *per task* may be much higher